### PR TITLE
Fix for issue #1269 - DataTable doesn't update filtered results when source data changes

### PR DESCRIPTION
**Pull request by ats17**

### CODE REVIEW

**Comment by cagataycivici** [comment]
This is likely to cause an infinite loop, see;

#1170

Please try this with a lazy loading datatable first.

**Comment by ats17** [comment]
Good point, I missed that and the other related issues. I've added a check to skip the filter call when lazy loading since in that case the data source would be handling the filtering.

My original commit checked filteredValue, but that would miss cases where nothing passed the filter before the data was updated. Now I'm checking hasFilter(). 

To avoid unneeded filtering, I've rewritten hasFilter() to only return true when at least one filter contains text. This looks to be compatible with all current callers.

File: components/datatable/datatable.ts [hunk 1]
```diff
@@ -1065,15 +1065,12 @@ export class DataTable implements AfterViewChecked,AfterViewInit,AfterContentIni
     }
 
     hasFilter() {
-        let empty = true;
-        for(let prop in this.filters) {
-            if(this.filters.hasOwnProperty(prop)) {
-                empty = false;
```
Comment (ats17) [inline components/datatable/datatable.ts]: To minimize changes, I probably should have simply added 
`if (this.filters[prop].value.length) `
in front of
`empty = false;`
in the original code.

**Comment by ats17** [comment]
I've made a mess out of this repo so I'm going to close this pull request and open a clean one

**Comment by cagataycivici** [comment]
Thanks, no need for a new PR, we'll do it at 1295

**Pull request closed without merging**